estate/models/estate_property.py

```diff
@@ -0,0 +1,115 @@
+    ]
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=lambda self: date.today() + timedelta(days=90))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer(string="Facade")
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[
+            ('north', "North"),
+            ('south', "South"),
+            ('east', "East"),
+            ('west', "West"),
+        ],
+        string="Garden Orientation"
+    )
+    available = fields.Boolean(default=True)
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    salesman_id = fields.Many2one("res.users", string="Salesman", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    state = fields.Selection(
+        selection=[
+            ('new', "New"),
+            ('offer_received', "Received"),
+            ('offer_accepted', "Accepted"),
+            ('sold', "Sold"),
+            ('cancelled', "Cancelled"),
+            ('refused', "Refused")
+        ],
+        required=True,
+        copy=False,
+        default='new'
+    )
+    note = fields.Text(string="Special mention about the property.")
+    total_area = fields.Integer(string="Total Area (sqm)", compute="_compute_total_area")
+    best_offer = fields.Float(string="Best Offer", compute="_compute_best_price")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_price(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_offer = max(record.offer_ids.mapped("price"))
+            else:
+                record.best_offer = 0.0
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = None
+
+    def cancel_property_button(self):
+        for record in self:
+            if record.state != 'sold':
+                record.state = 'cancelled'
+            else:
+                raise UserError("You cannot cancel a sold property.")
```

Comment:
Most strings are automatically made translatable: field names, module and model names, etc. But this is not the case for error messages. To make them translatable, use the `_()` method like so
```suggestion
                raise UserError(_("You cannot cancel a sold property."))
```